misc.c

```diff
@@ -308,28 +308,50 @@ char *convert_to_unix_name(struct cifsd_share_config *share, char *name)
 	return new_name;
 }
 
-char *cifsd_convert_dir_info_name(struct cifsd_dir_info *d_info,
-				  const struct nls_table *local_nls,
-				  int *conv_len)
+/**
+ * convname_updatenextoffset() - convert name to UTF, update next_entry_offset
+ * @namestr:            source filename buffer
+ * @len:                source buffer length
+ * @size:               used buffer size
+ * @local_nls           code page table
+ * @name_len:           file name length after conversion
+ * @next_entry_offset:  offset of dentry
+ * @buf_len:            response buffer length
+ * @data_count:         used response buffer size
+ * @no_namelen_field:	flag which shows if a namelen field flag exist
+ *
+ * Return:      return error if next entry could not fit in current response
+ *              buffer, otherwise return encode buffer.
+ */
+char *convname_updatenextoffset(char *namestr, int len, int size,
+		const struct nls_table *local_nls, int *name_len,
+		int *next_entry_offset, int *buf_len, int *data_count,
+		int alignment, bool no_namelen_field)
```

Comment:
@namjaejeon Please, don't. That's a huge step backward, not forward.

-ss

Reply:
Yep, I know, we can clean-up it base on stable status with xfstests and smbtorture.